cython/CMakeLists.txt

```diff
@@ -45,4 +44,12 @@ if (GTSAM_INSTALL_CYTHON_TOOLBOX)
   install_cython_scripts("${PROJECT_SOURCE_DIR}/cython/gtsam" "${GTSAM_CYTHON_INSTALL_PATH}" "*.py")
   install_cython_scripts("${PROJECT_SOURCE_DIR}/cython/gtsam_unstable" "${GTSAM_CYTHON_INSTALL_PATH}" "*.py")
 
+  # Adding custom function here so that gtsam_eigency is installed before
```

Comment:
I really think that this should be a separate target than the normal cython build target, so to install people can do `make cython_install` or something, and it's easier and clearer if you want to add a option to auto install.

Reply:
So this only runs if people do `make install`. The build step is completely handled via `make`.

Reply:
So, I thought that we agreed per our discussion that this should *not* run if people make install. But install should run if `make cython_install` is called and build is out of date.

Reply:
Currently that is not possible since CMake doesn't support adding dependencies on install commands. That's an open issue on the CMake repository.

Using make install allows CMake to manage the outdated dependencies directly this way, while also truly installing the python package. It didn't make sense to me to have make install copy to an install path and then a separate install package step.

If you would rather have the separate install command, then we can't have CMake manage the build dependencies at the moment.

Reply:
Yes, I don’t think we should change behavior, and I think there should be a separate command for invoking python setup

Reply:
In that case then we have no way of enforcing CMake to rebuild  when something happens.

I really don't understand why this is a problem. Currently the status quo is that we do make install and then we have to manually go to the cython folder to run the setup file. All I did was automate that last step as part of the make install process, so there's nothing changed from the previous way of doing things. We didn't have an extra command then and we don't have it now.

Reply:
Maybe we should chat in person - python install is frought with issue, and I think making it part of "make install" will create more problems than it will solve.

Reply:
So after a good(?) night's sleep, I actually figured out a way to accomplish the exact requirement. Have a separate build target yet have it automatically managed by cmake.

The idea is to use the `${PROJECT_BINARY_DIR}/cython` folder directly to hold everything and then `make python-install` will just use the `setup.py` there. This bypasses the whole "install" step, allowing us to achieve what we want. :grin: